docs/tutorials/agent_framework/RAG_with_conversational_flow_agent.md

```diff
@@ -0,0 +1,700 @@
+
+```
+
+## 2. Prepare LLM
+
+Find more details: [Remote model](https://opensearch.org/docs/latest/ml-commons-plugin/remote-models/index/)
+
+We use [Bedrock Claude model](https://aws.amazon.com/bedrock/claude/) in this tutorial. You can also use other LLM.
+
+1. Create connector
+```
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "BedRock Claude instant-v1 Connector ",
+  "description": "The connector to BedRock service for claude model",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+    "region": "us-east-1",
+    "service_name": "bedrock",
+    "anthropic_version": "bedrock-2023-05-31",
+    "max_tokens_to_sample": 8000,
+    "temperature": 0.0001,
+    "response_filter": "$.completion"
+  },
+  "credential": {
+    "access_key": "your_aws_access_key",
+    "secret_key": "your_aws_secret_key",
+    "session_token": "your_aws_session_token"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-instant-v1/invoke",
+      "headers": {
+        "content-type": "application/json",
+        "x-amz-content-sha256": "required"
+      },
+      "request_body": "{\"prompt\":\"${parameters.prompt}\", \"max_tokens_to_sample\":${parameters.max_tokens_to_sample}, \"temperature\":${parameters.temperature},  \"anthropic_version\":\"${parameters.anthropic_version}\" }"
+    }
+  ]
+}
+```
+
+Copy the connector id from the response. 
+
+2. register model
+
+```
+POST /_plugins/_ml/models/_register
+{
+    "name": "Bedrock Claude Instant model",
+    "function_name": "remote",
+    "description": "Bedrock Claude instant-v1 model",
+    "connector_id": "your_LLM_connector_id"
+}
+```
+Copy the LLM model id from the response, will use it in following steps.
+
+3. Deploy model
+```
+POST /_plugins/_ml/models/your_LLM_model_id/_deploy
+```
+
+4. Test predict
+```
+POST /_plugins/_ml/models/your_LLM_model_id/_predict
+{
+  "parameters": {
+    "prompt": "\n\nHuman: how are you? \n\nAssistant:"
+  }
+}
+```
+
+## 3. Create Agent
+Agent framework provides several agent types: `flow`, `conversational_flow` and `conversational`.
+
+We will use `conversational_flow` agent in this tutorial.
+
+The agent consists of:
+1. meta info: `name`, `type`, `description`
+2. `app_type`: this is to differentiate different application type
+3. `memory`: this is to store agent execution result, so user can retrieve memory later and continue one conversation.
```

Comment:
this is to store the user's conversations including the user's questions and the LLM's results. So an agent can retrieve chat history of conversation from the memory to use for the interactions with LLM.

Reply:
done with Fanit's comment